internal/keycloakb/passwordgeneration.go

```diff
@@ -0,0 +1,100 @@
+package keycloakb
+
+import (
+	"math/rand"
+	"strconv"
+	"strings"
+	"time"
+	"unicode"
+)
+
+// GeneratePassword generates a password of a given length
+func GeneratePassword(minLength int) string {
+
+	var pwdElems []string
+	const alphabet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ?!*0123456789"
+
+	for j := 0; j < minLength; j++ {
+		pwdElems = append(pwdElems, string(alphabet[rand.Intn(len(alphabet))]))
+	}
+
+	pwd := strings.Join(pwdElems, "")
+	return pwd
+}
+
+// GeneratePasswordFromKeycloakPolicy generates a random password respecting the keycloak password policy
+func GeneratePasswordFromKeycloakPolicy(policy string, minLength int) (string, error) {
```

Comment:
minLength parameter is not used. You might consider removing it or using the max value of this minLength and the one converted from the parameter length(...) provided in the given policy

Reply:
DONE